MRFIXIT: rockchip adds the feature of defining a custom resolution in batocera.conf for n64. Pull this resolution and use it in the config. Also calculate the proper 4/3 screen width to be used and set the new left-offset value that I patched in to properly center the window. This avoids the game from being stretched to the full 16/9 resolution.
--- a/configgen/generators/mupen/mupenConfig.py
+++ b/configgen/generators/mupen/mupenConfig.py
@@ -23,8 +23,25 @@
 	if not iniConfig.has_section("Video-General"):
 		iniConfig.add_section("Video-General")
 	iniConfig.set("Video-General", "Version", "1")
-	iniConfig.set("Video-General", "ScreenWidth",  gameResolution["width"])
-	iniConfig.set("Video-General", "ScreenHeight", gameResolution["height"])
+
+	ra_width = gameResolution["width"]
+	ra_height = gameResolution["height"]
+	leftoffset = 0
+	ra_res = subprocess.Popen(['/usr/bin/batocera-settings', '-command', 'load', '-key','n64.resolution'], stdout=subprocess.PIPE).communicate()[0].split("x")
+	if unicode(ra_res[0]).isnumeric():
+		_width = int(ra_res[0].replace("\n",""))
+		if len(ra_res) > 1:
+			if unicode(ra_res[1].replace("\n","")).isnumeric():
+				ra_height = int(ra_res[1].replace("\n",""))
+
+	ra_width = int(ra_height * (4.0/3.0))
+	leftoffset = int((_width - ra_width) / 2)
+
+	if system.config['core'] == "gliden64":
+		ra_width = _width
+
+	iniConfig.set("Video-General", "ScreenWidth",  ra_width)
+	iniConfig.set("Video-General", "ScreenHeight", ra_height)
 	iniConfig.set("Video-General", "VerticalSync", "True")
 	
 	# wide screen mode
@@ -47,7 +61,7 @@
 	else:
 		iniConfig.set("Video-GLideN64", "AspectRatio", "1")
 
-    # Rice
+	# Rice
 	if not iniConfig.has_section("Video-Rice"):
 			iniConfig.add_section("Video-Rice")
 
@@ -61,6 +75,10 @@
 		iniConfig.set("Video-Rice",       "ShowFPS",  "False")
 		iniConfig.set("Video-Glide64mk2", "show_fps", "8")
 
+	# Left Offset
+	iniConfig.set("Video-Rice", "ScreenLeftOffset", leftoffset)
+	iniConfig.set("Video-Glide64mk2", "ScreenLeftOffset", leftoffset)
+
         # custom : allow the user to configure directly mupen64plus.cfg via batocera.conf via lines like : n64.mupen64plus.section.option=value
         for user_config in system.config:
                 if user_config[:12] == "mupen64plus.":

